### PR TITLE
Enable Default Hyperparamenter with Ray Tune

### PR DESCRIPTION
**Pull request by yinweisu**
*Issue #, if available:*
https://github.com/awslabs/autogluon/issues/2068

*Description of changes:*
* Added the ability to use default hyperparameter when doing HPO with ray tune.

Example log with `NN_TORCH` and `FASTAI`, which use ray tune as backend. You can see default hyperparameters are correctly used by the single trial. Not shown by the log but same applies to multiple trials, where the first trial would use the default hyperparameters.
```python
Fitting 2 L1 models ...
Hyperparameter tuning model: NeuralNetFastAI ... Tuning model for up to 44.96s of the 99.9s of remaining time.
(model_trial pid=14381) {'layers': None, 'emb_drop': 0.1, 'ps': 0.1, 'bs': 256, 'lr': 0.01, 'epochs': 30, 'early.stopping.min_delta': 0.0001, 'early.stopping.patience': 20, 'smoothing': 0.0}
Fitted model: NeuralNetFastAI/2a4db_00000 ...
        0.898    = Validation score   (roc_auc)
        1.79s    = Training   runtime
        0.02s    = Validation runtime
Hyperparameter tuning model: NeuralNetTorch ... Tuning model for up to 44.96s of the 91.47s of remaining time.
(model_trial pid=14734) {'num_epochs': 500, 'epochs_wo_improve': 20, 'activation': 'relu', 'embedding_size_factor': 1.0, 'embed_exponent': 0.56, 'max_embedding_dim': 100, 'y_range': None, 'y_range_extend': 0.05, 'dropout_prob': 0.1, 'optimizer': 'adam', 'learning_rate': 0.0003, 'weight_decay': 1e-06, 'proc.embed_min_categories': 4, 'proc.impute_strategy': 'median', 'proc.max_category_levels': 100, 'proc.skew_threshold': 0.99, 'use_ngram_features': False, 'num_layers': 2, 'hidden_size': 128, 'max_batch_size': 512, 'use_batchnorm': False, 'loss_function': 'auto'}
Fitted model: NeuralNetTorch/2f278_00000 ...
        0.8924   = Validation score   (roc_auc)
        2.21s    = Training   runtime
        0.02s    = Validation runtime
Fitting model: WeightedEnsemble_L2 ... Training model for up to 99.9s of the 87.22s of remaining time.
        0.9183   = Validation score   (roc_auc)
        0.18s    = Training   runtime
        0.0s     = Validation runtime
AutoGluon training complete, total runtime = 12.97s ... Best model: "WeightedEnsemble_L2"
TabularPredictor saved. To load, use: predictor = TabularPredictor.load("AutogluonModels/ag-20220914_182927/")
```
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: core/src/autogluon/core/hpo/ray_hpo.py [hunk 1]
```diff
@@ -325,13 +335,21 @@ def _validate_resources_per_trial(resources_per_trial):
 def _convert_search_space(search_space: dict):
     """Convert the search space to Ray Tune search space if it's AG search space"""
     tune_search_space = search_space.copy()
+    default_hyperparameters = dict()
     for hyperparmaeter, space in search_space.items():
         if isinstance(space, Space):
             tune_search_space[hyperparmaeter] = RaySpaceConverterFactory.get_space_converter(space.__class__.__name__).convert(space)
-    return tune_search_space
+            default_hyperparameters[hyperparmaeter] = space.default
```
Comment (Innixma) [inline core/src/autogluon/core/hpo/ray_hpo.py:342]: nit: typo on hyperparameter

File: core/src/autogluon/core/hpo/ray_hpo.py [hunk 2]
```diff
@@ -216,9 +216,19 @@ def run(
...
-    search_space = _convert_search_space(search_space)
+    search_space, default_hyperparameters = _convert_search_space(search_space)
+
+    searcher = _get_searcher(
+        hyperparameter_tune_kwargs,
+        metric,
+        mode,
+        default_hyperparameters=default_hyperparameters,
+        supported_searchers=ray_tune_adapter.get_supported_searchers()
+    )
```
Comment (Innixma) [inline core/src/autogluon/core/hpo/ray_hpo.py:227]: nit: use arg name hints here for all args, just a bit easier to follow

File: core/src/autogluon/core/hpo/ray_hpo.py [hunk 3]
```diff
@@ -216,9 +216,19 @@ def run(
...
+    scheduler = _get_scheduler(
+        hyperparameter_tune_kwargs,
+        supported_schedulers=ray_tune_adapter.get_supported_schedulers()
+    )
```
Comment (Innixma) [inline core/src/autogluon/core/hpo/ray_hpo.py:231]: ditto nit: use arg name hints for all args

**Review by Innixma — COMMENTED** [review]
Nice! Excited about this improvement, it should be quite impactful and might allow us to switch all our HPO to ray-based.

Added some comments, also CI seems to be failing with ray related errors.

**Review by Innixma — APPROVED** [review]
LGTM! Would recommend checking what happens when default config is outside the range of the search space.

**Comment by yinweisu** [comment]
Tried default hyperparameter out of the space. Worked without error, and the following hyperparameters being generated are still in bound

**Comment by Innixma** [comment]
Great! Sounds good, feel free to merge

**Pull request merged**